frontend/apps/reader/modules/readerrolling.lua

```diff
@@ -1310,6 +1364,14 @@ Note that %1 (out of %2) xpaths from your bookmarks and highlights have been nor
     })
 end
 
+function ReaderRolling:onToggleHideNonlinear()
+    self.hide_nonlinear = not self.hide_nonlinear
+    self.ui.document:setHideNonlinear(self.hide_nonlinear)
+    self.ui.document:render()
```

Comment:
Nowhere we need to call `self.ui.document:render()` directly (the fact that you included the setting in the global doc properties in crengine will have it notice a re-rendering is needed next time... something requests something :)
Usually, we have ReaderRolling:onUpdatePos() called instead.